stbt/keyboard.py

```diff
@@ -327,3 +337,24 @@ def _add_weights(G):
                 # take a shortcut through here.
                 for target in targets:
                     G[node][target]["weight"] = 100
+
+
+def _join_with_commas(items, last_one=", "):
+    """
+    >>> _join_with_commas(["A", "B", "C"], last_one=" or ")
+    'A, B or C'
+    >>> _join_with_commas(["A", "C"], last_one=" or ")
+    'A or C'
+    >>> _join_with_commas(["A"], last_one=" or ")
+    'A'
+    >>> _join_with_commas([], last_one=" or ")
+    ''
+    """
+    if len(items) > 1:
+        return last_one.join([
+            ", ".join(items[:-1]),
+            items[-1]])
+    elif len(items) == 1:
+        return items[0]
+    else:
+        return ""
```

Comment:
A less readable oneliner:

    return " or ".join([", ".join(items[:-1]), items[-1]]) if items else ""

Reply:
```
>>> items = ['a']
>>> " or ".join([", ".join(items[:-1]), items[-1]]) if items else ""
' or a'
```

Reply:
And yes, I made exactly that mistake in my initial implementation. :-)